drivers/sensor/adt7420/adt7420.c

```diff
@@ -0,0 +1,202 @@
+			   reg, (u8_t *) val, 2) < 0) {
+		return -EIO;
+	}
+
+	*val = sys_be16_to_cpu(*val);
+
+	return 0;
+}
+
+static int adt7420_temp_reg_write(struct adt7420_data *drv_data,
+			    u8_t reg, s16_t val)
+{
+	u8_t buf[3] = {reg, val >> 8, val & 0xFF};
+
+	return i2c_write(drv_data->i2c, buf, sizeof(buf), drv_data->i2c_addr);
+}
+
+static int adt7420_attr_set(struct device *dev,
+			   enum sensor_channel chan,
+			   enum sensor_attribute attr,
+			   const struct sensor_value *val)
+{
+	struct adt7420_data *drv_data = dev->driver_data;
+	u8_t val8, reg = 0;
+	u16_t rate;
+	s64_t value;
+
+	if (chan != SENSOR_CHAN_AMBIENT_TEMP) {
+		return -ENOTSUP;
+	}
+
+	switch (attr) {
+	case SENSOR_ATTR_SAMPLING_FREQUENCY:
+		rate = val->val1 * 1000 + val->val2 / 1000; /* rate in mHz */
+
+		switch (rate) {
+		case 240:
+			val8 = ADT7420_OP_MODE_CONT_CONV;
+			break;
+		case 1000:
+			val8 = ADT7420_OP_MODE_1_SPS;
+			break;
+		default:
+			return -ENOTSUP;
+		}
+
+		if (i2c_reg_update_byte(drv_data->i2c, drv_data->i2c_addr,
+					ADT7420_REG_CONFIG,
+					ADT7420_CONFIG_OP_MODE(~0),
+					ADT7420_CONFIG_OP_MODE(val8)) < 0) {
+			SYS_LOG_DBG("Failed to set attribute!");
+			return -EIO;
+		}
+
+		return 0;
+	case SENSOR_ATTR_UPPER_THRESH:
+		reg = ADT7420_REG_T_HIGH_MSB;
+		/* Fallthrough */
+	case SENSOR_ATTR_LOWER_THRESH:
+		if (!reg)
+			reg = ADT7420_REG_T_LOW_MSB;
+
+		value = (s64_t)val->val1 * 1000000 + val->val2;
+		value = (value / ADT7420_TEMP_SCALE) << 1;
+
+		if (adt7420_temp_reg_write(drv_data, reg, value) < 0) {
+			SYS_LOG_DBG("Failed to set attribute!");
+			return -EIO;
+		}
+
+		return 0;
+	default:
+		return -ENOTSUP;
+	}
+
+	return 0;
+}
+
+static int adt7420_sample_fetch(struct device *dev, enum sensor_channel chan)
+{
+	struct adt7420_data *drv_data = dev->driver_data;
+	s16_t value;
+
+	__ASSERT_NO_MSG(chan == SENSOR_CHAN_ALL ||
+			chan == SENSOR_CHAN_AMBIENT_TEMP);
+
+	if (adt7420_temp_reg_read(drv_data, ADT7420_REG_TEMP_MSB, &value) < 0)
+		return -EIO;
+
+	drv_data->sample = value >> 1; /* use 15-bit only */
+
+	return 0;
+}
+
+static int adt7420_channel_get(struct device *dev,
+		enum sensor_channel chan,
+		struct sensor_value *val)
+{
+	struct adt7420_data *drv_data = dev->driver_data;
+	s32_t value;
+
+	if (chan != SENSOR_CHAN_AMBIENT_TEMP) {
+		return -ENOTSUP;
+	}
+
+	value = (s32_t)drv_data->sample * ADT7420_TEMP_SCALE;
+	val->val1 = value / 1000000;
+	val->val2 = value % 1000000;
+
+	return 0;
+}
+
+static const struct sensor_driver_api adt7420_driver_api = {
+	.attr_set = adt7420_attr_set,
+	.sample_fetch = adt7420_sample_fetch,
+	.channel_get = adt7420_channel_get,
+#ifdef CONFIG_ADT7420_TRIGGER
+	.trigger_set = adt7420_trigger_set,
+#endif
+};
+
+static int adt7420_probe(struct device *dev)
+{
+	struct adt7420_data *drv_data = dev->driver_data;
+	u8_t value;
+	int ret;
+
+	ret = i2c_reg_read_byte(drv_data->i2c, drv_data->i2c_addr,
+				ADT7420_REG_ID, &value);
+	if (ret)
+		return ret;
```

Comment:
Please add braces after all the if statements.
http://docs.zephyrproject.org/contribute/contribute_guidelines.html#coding-style

Reply:
sure